config/nrfconnect/chip-module/Kconfig.defaults

```diff
@@ -85,6 +85,11 @@ config NEWLIB_LIBC_NANO
     bool
     default y
 
+# Warn a user if memory left for the heap is too small
+config NEWLIB_LIBC_MIN_REQUIRED_HEAP_SIZE
+    int
+    default 16384
```

Comment:
How was this value determined?

Reply:
I remember that when testing HeapHighWatermark attribute (maximum heap usage so far) it's never reached 10kB. So 16kB looks quite safe to me. Or at least it's safer than the default 2kB. We can think about some automated tests to verify if that holds.